### PR TITLE
Replaced inputs on the left and swapped out overview and scatterplot modules

### LINKED ISSUE #138: Replace filters + card textOutputs with new data model
## Feature Details
Within the `ui.R` and `server.R` file, we want to modify the selectInput options and the card content with the new data model.

image

### LINKED ISSUE #139: Update the Overview modules and supplemental functions
## Feature Details
On the Overview Tab, we want to:

1. Replace `Overview_Table` with `Widget_GroupOverview` in the `mod_study_overview_ui` and `mod_study_overview_server`.
2. Modify the `textOutput` that produce the KRI counts with the new data model
3. Replace the scatterplots with the new widgets

We will add the new data model to the `server.R` file as needed.

image

### PR DESCRIPTION
**Pull request by taylorrodgers**
I modified the left hand card to use the new data model, as well as modified the selectInputs.

I also swapped out both the table and scatterplot found on the overview tab with the new widgets.

Closes #138 
Closes #139

Note: I partially finished #143, although I'll still leave it open until we resolve some lingering issues with action clicks. These will require some additional work in gsm and also require the completion of more tabs. For that reason, I commented out most changes to continue work no the next pull request.

### CODE REVIEW

**Comment by jonthegeek** [comment]
Pulling comments over from Teams:

The app corresponding to this PR is live at https://openrbqm.shinyapps.io/gsmApp-pr-144/ !

@taylorrodgers : to make sure I understand what we should expect to see: 
 
- The filter panel on the left should work (with data in the gsm v2 format) (#138).
- The filter panel should only show fields relative to the current tab (#143).
- The Overview Table on the Study Overview tab should display data for the 2 metrics we have in the sample data.
- The Scatter Plots on Study Overview should also display for those 2 metrics.
- Other tabs are expected to NOT work because we haven't finished updating them to gsm v2.

Does that sound right? I don't want to report broken things if you don't expect them to work 🙃

I'll move forward with the rest of the review assuming those things are true. That all makes sense, but I'm new to this package, so I want to make sure I didn't fundamentally miss anything!

**Comment by jonthegeek** [comment]
I explicitly "hid" the functions that we aren't using right now (by commenting them out and adding `zzz-` to the filenames). That way they don't slow things down nor show up in coverage reports, but the code is there when we revise/rewrite that module.

File: R/server.R
Comment (jonthegeek) [inline R/server.R:95]: Let's be consistent about naming. `study_overview_server()` and `modScatterServer()` are using different naming schemes. Their filenames follow the same pattern, but not the function names. Most functions appear to follow the (somewhat more standard for R) `snake_case` format, so let's standardize around that (without the `mod_` piece, too, since the others don't use it). So this should be `scatter_server`, and should be updated in its definition in `mod_scatter_server.R` as well.

File: R/run_app.R
Comment (jonthegeek) [inline R/run_app.R:16]: I don't think we should give these default values. I made some readability edits, but, as I get deeper into this, these should either have default values, should be required, or should really be `NULL`able (where the `NULL` really means "I am not supplying a value and none should be used`). See [Hadley's in-progress Tidy Design Principles book](https://design.tidyverse.org/inputs-explicit.html) for some tips on how to make functions easier to follow via careful default args.

More specifically, while we'll use `gsm::reportingResults` in examples, the app should fail to launch if the user doesn't supply that argument, so there shouldn't be a default value. `dfBounds` is probably `NULL`-able (because it looks like we ignore it downstream if it isn't supplied), but the app should fail if any of the others are missing.

File: R/run_app.R
Comment (jonthegeek) [inline R/run_app.R:27]: Our function should take `dfResults`, `dfGroups`, `dfMetrics`, and `dfBounds`, and return a `server` function that takes `input, output, session` as args. That will make the calling here more straightforward/easier to understand.

I'm going to separate (most? all?) of these things about "how I wish things were structured" out into issues so we can discuss them there without blocking this PR from being merged.

**Review by jonthegeek — CHANGES_REQUESTED** [review]
I have some comments in here that will be pulled out into separate issues; I'll close those comments as I pull them out. There are some real ones left beyond that, but overall this is a great start!

However, let's figure out what in `inst/www/style.css` is interfering with the widget appearance (as discussed somewhat on Teams). The never version of tab 1 should look "right" before we merge this PR, even if there are other things to make more-ideal.

File: R/run_app.R
Comment (jonthegeek) [reply R/run_app.R:27]: Split into #149.

File: R/run_app.R
Comment (jonthegeek) [reply R/run_app.R:16]: I didn't split this off into a separate issue, since we're defining the usage here. I don't think we should do this "use the sample data if they don't supply data" idea, even if we don't get these "perfect" just yet.

File: R/server.R
Comment (taylorrodgers) [reply R/server.R:27]: I'm unsure as well. I am keeping it until I can verify we no longer need it.

I think it's fine if we were to make a ticket for after we finish the metric details tab to remove it if possible.

File: R/server.R
Comment (taylorrodgers) [reply R/server.R:1]: The two observe functions would be the most ideal to move to a separate R script.

File: R/server.R
Comment (taylorrodgers) [reply R/server.R:95]: Are you wanting that done on this PR or will this be a ticket?

File: R/run_app.R
Comment (taylorrodgers) [reply R/run_app.R:16]: Let me re-phrase this to make sure I understand:

- We don't think we should have `dfResults = NULL` in the function deftauls
- We don't think we should load the sample data via the run_app function
- We should only include the sample data in the example of the function documentation

If that's the case, would you be fine with having this as a ticket for later?

Those sample data sets in `run_app` make it easier for me to test changes iteratively. I think it would be good to have a reminder to remove it later as we are closer towards completion.

File: R/server.R
Comment (taylorrodgers) [reply R/server.R:95]: I went ahead and make a ticket for this. I don't recall how `modScatterServer` made its way in there. There were different developers over time, so some of the naming conventions were adopted inconsistently.

File: R/server.R
Comment (taylorrodgers) [reply R/server.R:1]: I see your ticket. Not sure yet if it'll be in the next PR, but I think it could certainly be accomplished by next week.

**Comment by taylorrodgers** [comment]
@jonthegeek, I went ahead and removed the css entries that were interfering with the new overview table.

File: R/run_app.R
Comment (samussiah) [reply R/run_app.R:16]: Maybe add a development parameter that defaults to `FALSE` and pulls in testing data otherwise.

File: R/server.R
Comment (jonthegeek) [reply R/server.R:27]: Logged in #154

File: R/server.R
Comment (jonthegeek) [reply R/server.R:95]: #152

File: R/run_app.R
Comment (jonthegeek) [reply R/run_app.R:16]: This will work for now! As I sort out the way we ingest data, we might have a version that accepts reactives, which makes things even more complicated... so let's not try to shoehorn this fix in here!

**Review by jonthegeek — APPROVED** [review]

**Pull request merged**
